exp/views/mixins.py

```diff
@@ -39,6 +42,34 @@ def dispatch(self, request, *args, **kwargs):
             )
 
 
+class StudyLookupMixin:
+
+    study = None
+
+    def get_study(self):
```

Comment:
Is there a reason we can't use `property` or `functools.cached_property` here?

Reply:
I didn't know about `cached_property` - very cool, changed this!